pkg/snapshot/copier/copier_test.go

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2018 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package copier_test
+
+import (
+	"os"
+
+	"github.com/gardener/etcd-backup-restore/pkg/miscellaneous"
+	. "github.com/gardener/etcd-backup-restore/pkg/snapshot/copier"
+	brtypes "github.com/gardener/etcd-backup-restore/pkg/types"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var (
+	copierConfig    *brtypes.CopierConfig
+	snapstoreConfig *brtypes.SnapstoreConfig
+	ss              brtypes.SnapStore
+	ds              brtypes.SnapStore
+	copier          *Copier
+)
+
+var _ = Describe("Running Copier", func() {
+	BeforeEach(func() {
+		copierConfig = NewCopierConfig()
+		copierConfig.SourceSnapstore = &brtypes.SnapstoreConfig{
+			MaxParallelChunkUploads: 5,
+			TempDir:                 "/tmp",
+			Provider:                "Local",
+			Container:               targetSnapstoreDir,
+		}
+		snapstoreConfig = &brtypes.SnapstoreConfig{
+			MaxParallelChunkUploads: 5,
+			TempDir:                 "/tmp",
+			Provider:                "Local",
+			Container:               snapstoreDir,
+		}
+
+		var err error
+		ds, ss, err = GetSourceAndDestinationStores(copierConfig, snapstoreConfig)
+		Expect(err).ToNot(HaveOccurred())
+		copier = NewCopier(ss, ds, logger)
+	})
+	It("should run the copy command succesfully", func() {
+		fullSourceStoreSnapshot, deltaSrourceStoreSnapList, err := miscellaneous.GetLatestFullSnapshotAndDeltaSnapList(ss)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(copier.Run()).ToNot(HaveOccurred())
+		fullTargetStoreSnapshot, deltaTargetStoreSnapList, err := miscellaneous.GetLatestFullSnapshotAndDeltaSnapList(ss)
```

Comment:
```suggestion
		fullTargetStoreSnapshot, deltaTargetStoreSnapList, err := miscellaneous.GetLatestFullSnapshotAndDeltaSnapList(ds)
```

Should the listing here be from the destination snapstore?

Reply:
Ah yes, just a copy paste mistake :/ 
Also this won't work, because the `Prefix` is different for `source` and `target`  so I will implement something else.